lib/prm/doctor_api/doctor_search.ex

```diff
@@ -0,0 +1,46 @@
+defmodule PRM.DoctorSearch do
+  @moduledoc false
+
+  use Ecto.Schema
+
+  import Ecto.Changeset
+
+  schema "doctor_search" do
+    field :first_name, :string
+    field :last_name, :string
+    field :second_name, :string
+    field :speciality, :string
+    field :region, :string
+    field :area, :string
+    field :msp_id, :string
+    field :is_active, :boolean, default: true
+  end
+
+  def validate(attrs \\ %{}) do
+    fields = ~W(
+      first_name
+      last_name
+      second_name
+      speciality
+      region
+      area
+    )a
+
+    %PRM.DoctorSearch{}
+    |> cast(attrs, fields)
+    |> validate_any_is_present(fields)
+  end
+
+  def validate_any_is_present(changeset, fields) do
+    if Enum.any?(fields, &present?(changeset, &1)) do
+      changeset
+    else
+      add_error(changeset, hd(fields), "One of these fields must be present: #{inspect fields}")
```

Comment:
In previous request it was. 
I'll revert validation message for each required_any field, but I think it's better to tell that
`:required, "This field is required only when any of %fields% fields are not present"`